Framework/Algorithms/test/GetEiTest.h

```diff
@@ -271,5 +270,113 @@ class GetEiTest : public CxxTest::TestSuite {
     return alg;
   }
 };
+class GetEiTestPerformance : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static GetEiTestPerformance *createSuite() {
+    return new GetEiTestPerformance();
+  }
+  static void destroySuite(GetEiTestPerformance *suite) { delete suite; }
+
+  void setUp() {
+    inputWS1 = createTestWorkspaceWithMonitors();
+    outputName1 = "eitest1";
+    AnalysisDataService::Instance().add(outputName1, inputWS1);
+
+    inputWS2 = createTestWorkspaceWithMonitors();
+    outputName2 = "eitest2";
+    AnalysisDataService::Instance().add(outputName2, inputWS1);
+
+    // This algorithm needs a name attached to the workspace
+  }
+
+  void tearDown() {
+    AnalysisDataService::Instance().remove(outputName1);
+    AnalysisDataService::Instance().remove(outputName2);
+  }
+
+  void test_Result_For_Good_Estimate() {
+    const double input_ei = 15.0;
+    const bool fixei = false;
+
+    IAlgorithm_sptr alg;
+    TS_ASSERT_THROWS_NOTHING(
+        alg = runGetEiUsingTestMonitors(outputName1, input_ei, fixei));
+  }
+
+  void test_Result_When_Fixing_Ei() {
+    const double input_ei = 15.0;
+    const bool fixei = true;
+
+    IAlgorithm_sptr alg;
+    TS_ASSERT_THROWS_NOTHING(
+        alg = runGetEiUsingTestMonitors(outputName2, input_ei, fixei));
+  }
+
+private:
+  /// nearly the same method as the unit test above
+  /// changed from std::string to MatrixWorkspace_sptr
+  IAlgorithm_sptr runGetEiUsingTestMonitors(const std::string inWSName,
+                                            const double energyGuess,
+                                            const bool fixei) {
+
+    IAlgorithm_sptr alg =
+        AlgorithmManager::Instance().createUnmanaged("GetEi", 2);
+    alg->initialize();
+    alg->setPropertyValue("InputWorkspace", inWSName);
+    alg->setProperty("Monitor1Spec", 1);
+    alg->setProperty("Monitor2Spec", 2);
+    alg->setProperty("FixEi", fixei);
+    alg->setProperty("EnergyEstimate", energyGuess);
+    alg->setRethrows(true);
+    alg->execute();
+
+    return alg;
+  }
+
+  /// Same method as the unit test above
+  MatrixWorkspace_sptr
+  createTestWorkspaceWithMonitors(const bool includePeaks = true) {
```

Comment:
Is this just copied from the other test? Such duplication should be avoided if possible. Can it be extracted into a free function?